### PR TITLE
NIAD-0: Add description template to all pull requests.

### PR DESCRIPTION
**Pull request by matthewdoreilly**
* Add description template to all pull requests
  * See how this template would look below
* Update workflow so task or hotfix branches can be named
  * For example, this branch is more of a task than a feature
  * I propose we leave the `NIAD-[0-9]` prefix on PR titles and branch names so we enforce that. And any branch which isn't tied to a ticket (like this one) we have the prefix something like `NIAD-0`
* Updated README

---

## Description

Please explain the changes you have made here.

## Jira Ticket

Please add a link to the related Jira ticket (if any).

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [ ] Acceptance Criteria met
- [ ] Commit messages are meaningful
- [ ] Manually tested
- [ ] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [ ] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [ ] Add/update any relevant documentation

### CODE REVIEW

**Review by saileshp-kainos — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## Description
+
+Please briefly explain the changes you made here.
```
Comment (andrewwa-kainos) [inline .github/pull_request_template.md:3]: I wouldn't say "briefly". PR descriptions should be thorough enough to explain the functionality and reasons for design choices made. I'd rather a PR author erred on the side of giving too much detail than too little. What do you think?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Jira Ticket
+
+Please add link to the Jira ticket.
+
+## Coding Standards
+
+https://gpitbjss.atlassian.net/wiki/spaces/NIA/pages/2108522539/Java+Coding+Standards
```
Comment (andrewwa-kainos) [inline .github/pull_request_template.md:11]: This doesn't feel helpful. As a team we should be following the coding standards, but this is just bloating PR descriptions. We'll end up just automatically ignoring it, like banner-blindness. Maybe this link should be given in the Development section of the README.md?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] Acceptance Criteria met
+- [ ] Branch named {feature|task|bug|hotfix}/{NIAD-*}
+- [ ] Commit messages are meaningful
+- [ ] Manually tested
+- [ ] All unit tests passing
+- [ ] All integration tests passing
+- [ ] Pulled latest main into feature branch
+- [ ] Code reviewed
+- [ ] If your pull request depends on any other, please link them in the description
+- [ ] main branch is passing/green on CI
```
Comment (andrewwa-kainos) [inline .github/pull_request_template.md:24]: Some of these we can see from the github checks (Branch name, unit tests passing, integration tests passing, code reviewed) and others can become outdated (pulled latest main, main branch is green). One suggestion for re-wording. Would also suggest a tick for reviewing our own code prior to asking others to do so, and a reminder for any documentation tasks:

```suggestion
- [ ] Acceptance Criteria met
- [ ] Commit messages are meaningful
- [ ] Manually tested
- [ ] All PR dependencies linked in the description
- [ ] Reviewed own code
- [ ] Any required documentation updates are complete
```

**Review by n-r-b — APPROVED** [review]

File: .github/pull_request_template.md (hunk 3 above)
Comment (n-r-b) [reply .github/pull_request_template.md:24]: I think keeping documentation up to date is important, so keen for that checkbox to be added

File: .github/pull_request_template.md (hunk 1 above)
Comment (matthewdoreilly) [reply .github/pull_request_template.md:3]: Good shout. Yeah happy with that

File: .github/pull_request_template.md (hunk 3 above)
Comment (matthewdoreilly) [reply .github/pull_request_template.md:24]: True. The GitHub checks obviously will never be removed, I included them in the checklist as it's still part of our "Ready to merge". But happy to take them out

File: .github/pull_request_template.md (hunk 2 above)
Comment (matthewdoreilly) [reply .github/pull_request_template.md]: Added to the README

**Review by saileshp-kainos — APPROVED** [review]

**Review by andrewwa-kainos — APPROVED** [review]

**Pull request merged**